### PR TITLE
Change serialization of Chronos NaiveDate

### PR DESCRIPTION
**Pull request by sporto**
https://github.com/graphql-rust/juniper/issues/150

`NaiveDate` should serialize / deserialize from "2018-01-02" instead of a full ISO date.

### CODE REVIEW

File: juniper/src/integrations/chrono.rs [hunk 1]
```diff
@@ -157,17 +150,16 @@ mod test {
...
+        naivedate_test("1996-12-19", 1996, 12, 19);
     }
 
-    #[test]
-    fn naivedate_from_input_value_with_fractional_seconds() {
-        naivedate_test("1996-12-19T16:39:57.005-08:00", 1996, 12, 19);
-    }
+    fn naivedate_serialization() {
+        let date = NaiveDate::from_ymd(2000, 1, 2);
+        // let value = date.serialize().unwrap();
```
Comment (sporto) [inline juniper/src/integrations/chrono.rs]: @theduke can you tell me what needs to be called in order to serialize a `NaiveDate`. So convert this to an `InputValue` using the `resolve` function. Thanks

Comment (LegNeato) [reply juniper/src/integrations/chrono.rs]: Something like https://github.com/graphql-rust/juniper/blob/master/juniper/src/integrations/url.rs#L23 ?

File: juniper/src/integrations/chrono.rs [hunk 2]
```diff
@@ -157,17 +151,14 @@ mod test {
...
-    }
-
-    #[test]
-    fn naivedate_from_input_value_with_z_timezone() {
-        naivedate_test("1996-12-19T16:39:57Z", 1996, 12, 19);
+    fn naivedate_serialization() {
+        let date = NaiveDate::from_ymd(2000, 1, 2);
+        let value = date.resolve();
+        let expected = Value::string("2000-01-02".to_string());
+        assert_eq!(value, expected);
```
Comment (sporto) [inline juniper/src/integrations/chrono.rs]: @LegNeato there is a test above that tests that given an input value it parses correctly and returns a NaiveDate. This https://github.com/graphql-rust/juniper/blob/master/juniper/src/integrations/url.rs#L23 looks similar.

I would like a test that given a NaiveDate it tests that it serialises correctly. Something that calls the `resolve` function to serialize. I haven't found an example test that do this.

I'm the app I'm building I'm getting a panic when trying to serialize NaiveDate, unsure what is wrong, but knowing how to test the serialization will help to find the issue.

File: juniper/src/integrations/chrono.rs [hunk 3]
```diff
@@ -157,17 +151,17 @@ mod test {
...
-        naivedate_test("1996-12-19T16:39:57.005-08:00", 1996, 12, 19);
-    }
+    fn naivedate_serialization() {
+        use types::base::GraphQLType;
 
-    #[test]
-    fn naivedate_from_input_value_with_z_timezone() {
-        naivedate_test("1996-12-19T16:39:57Z", 1996, 12, 19);
+        let date = NaiveDate::from_ymd(2000, 1, 2);
+        let value = date.resolve(&(), None, &executor::Executor);
```
Comment (sporto) [inline juniper/src/integrations/chrono.rs]: I figured out what to pass for the first two args, but don't know about the third.
Any help on this? Thanks

Comment (sporto) [reply juniper/src/integrations/chrono.rs]: @LegNeato @theduke can I get your help on this please? Thanks

Comment (LegNeato) [reply juniper/src/integrations/chrono.rs]: @sporto I did a PR to your PR with some integration tests. Feel free to structure them another way.

**Comment by LegNeato** [comment]
Please update the doc comment at the top of the file and add this to the changelog.

This is a potentially breaking change if someone is using a non-rust client or storing results...so we should probably bump to `0.10`. Thoughts @theduke?

**Comment by sporto** [comment]
@LegNeato this PR is ready for review. Thanks

**Comment by LegNeato** [comment]
@sportoi sadly I don't have commit privs so can't review and merge. We'll have to wait for @theduke or @mhallin.

**Comment by theduke** [comment]
@sporto thanks and sorry for the delay.

**Pull request merged**